pkg/network/ebpf/c/protocols/classification/protocol-classification.h

```diff
@@ -149,10 +149,12 @@ __maybe_unused static __always_inline void protocol_classifier_entrypoint(struct
 
     protocol_stack_t *protocol_stack = get_protocol_stack_if_exists(&classification_ctx->tuple);
 
-    if (is_fully_classified(protocol_stack) || is_protocol_layer_known(protocol_stack, LAYER_ENCRYPTION)) {
+    if (is_fully_classified(protocol_stack)) {
         return;
     }
 
+    bool encryption_layer_known = is_protocol_layer_known(protocol_stack, LAYER_ENCRYPTION);
```

Comment:
I originally just called this method below on line 199 but the 4.14 verifier gets upset for some reason. either way I think this is fine, we just want to make sure we don't anything more than the TLS classification steps in the case that we already know that layer. this way we can still evaluate the added handshake metadata without doing more than necessary